x-pack/plugins/infra/public/containers/with_options.tsx

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import moment from 'moment';
+import React from 'react';
+import { InfraMetricType, InfraPathType } from '../../common/graphql/types';
+import { InfraOptions, Omit } from '../lib/lib';
+
+interface WithOptionsInjectedProps {
+  options: InfraOptions;
```

Comment:
How about injection the options separately to increase flexibility (matching the proposed change above)?